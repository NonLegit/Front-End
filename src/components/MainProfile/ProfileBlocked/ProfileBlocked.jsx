import { useNavigate } from 'react-router-dom';
import PersonOffIcon from '@mui/icons-material/PersonOff';
import { Typography } from '@mui/material';
import {
  BlockedBox, BlockedButton, BlockedIcon, Icon, Image,
} from './styles';

/**
 * ProfileBlocked
 *
 * @component ProfileBlocked
 * @property {function} handleCont - continue to profile page
 * @property {string} username, - username

 * @returns {React.Component} ProfileBlocked
 */

function ProfileBlocked({ username, handleCont }) {
  const navigate = useNavigate();

  return (
    <BlockedBox>
      <Image src="/back.jpg" />
      <Icon src="https://www.redditstatic.com/avatars/defaults/v2/avatar_default_7.png" />
      <BlockedIcon size="small" sx={{ backgroundColor: 'red', color: 'white' }}>
        <PersonOffIcon fontSize="string" />
      </BlockedIcon>
      <Typography sx={{ fontWeight: 700, marginTop: 1, fontSize: 22 }}>
        u/
        {username}
        {' '}
        is blocked
      </Typography>
      <Typography variant="body2" sx={{ marginTop: 2, marginBottom: 4 }}>Are you sure you want to continue to their profile?</Typography>
      <BlockedButton variant="contained" onClick={handleCont}>Continue</BlockedButton>
      <BlockedButton variant="outlined" onClick={() => { navigate(-1); }}>Go Back</BlockedButton>
    </BlockedBox>
  );
}

export default ProfileBlocked;
