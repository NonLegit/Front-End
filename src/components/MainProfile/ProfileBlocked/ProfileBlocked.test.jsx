import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import ProfileBlocked from './ProfileBlocked';

test('test snapshot', async () => {
  const tree = renderer.create(
    <Router>
      <ProfileBlocked />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
