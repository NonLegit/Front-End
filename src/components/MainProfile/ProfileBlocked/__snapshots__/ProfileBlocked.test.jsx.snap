// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-5sy9qi"
>
  <img
    className="css-18012q2"
    src="/back.jpg"
  />
  <img
    className="css-1wj4lpp"
    src="https://www.redditstatic.com/avatars/defaults/v2/avatar_default_7.png"
  />
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-k9hm34-MuiButtonBase-root-MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeString css-821wyw-MuiSvgIcon-root"
      data-testid="PersonOffIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M8.65 5.82C9.36 4.72 10.6 4 12 4c2.21 0 4 1.79 4 4 0 1.4-.72 2.64-1.82 3.35L8.65 5.82zM20 17.17c-.02-1.1-.63-2.11-1.61-2.62-.54-.28-1.13-.54-1.77-.76L20 17.17zm1.19 4.02L2.81 2.81 1.39 4.22l8.89 8.89c-1.81.23-3.39.79-4.67 1.45-1 .51-1.61 1.54-1.61 2.66V20h13.17l2.61 2.61 1.41-1.42z"
      />
    </svg>
  </button>
  <p
    className="MuiTypography-root MuiTypography-body1 css-8wm45u-MuiTypography-root"
  >
    u/
     
    is blocked
  </p>
  <p
    className="MuiTypography-root MuiTypography-body2 css-10sm64x-MuiTypography-root"
  >
    Are you sure you want to continue to their profile?
  </p>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-19en2pd-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Continue
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-d2am4k-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Go Back
  </button>
</div>
`;
